### PR TITLE
[DUOS-2662][DUOS-2663][DUOS-2664][DUOS-2665][DUOS-2666][DUOS-2667] Support the branded data libraries

### PR DESCRIPTION
**Pull request by fboulnois**
### Addresses

https://broadworkbench.atlassian.net/browse/DUOS-2662
https://broadworkbench.atlassian.net/browse/DUOS-2663
https://broadworkbench.atlassian.net/browse/DUOS-2664
https://broadworkbench.atlassian.net/browse/DUOS-2665
https://broadworkbench.atlassian.net/browse/DUOS-2666
https://broadworkbench.atlassian.net/browse/DUOS-2667

### Summary

Adds the initial version of the Broad, DUOS, MGB, eLwazi, and My Institution data libraries.

----
Have you read [Terra's Contributing Guide](https://github.com/DataBiosphere/terra-ui/wiki/Contributor-Guide) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/pages/DatasetSearch.js [hunk 1]
```diff
@@ -3,10 +3,43 @@ import { useEffect, useState } from 'react';
...
+  const versions = {
+    '/datalibrary': {
+      query: null,
+      icon: duosIcon,
+      title: 'DUOS Data Library',
+    },
+    '/datalibrary_broad': {
+      query: {
+        'match': {
+          'submitter.institution.id': '150' // Broad Institute of MIT and Harvard
```
Comment (fboulnois) [inline src/pages/DatasetSearch.js]: Perhaps I should match on exact name instead since the ids might be different between environments.

Comment (rushtong) [reply src/pages/DatasetSearch.js]: Yes, the IDs are different across environments, let's go with name

**Review by rushtong — APPROVED** [review]
I really like this approach 👍🏽  One possible abstraction idea would be around `versions`. I could see that being an external configuration object passed into the component as a route prop instead of looking for it in the `DatasetSearch` component.

**Review by hams7504 — APPROVED** [review]
Looks great!

**Review by rushtong — APPROVED** [review]
Still looks good 👍🏽

**Pull request merged**